### PR TITLE
Add get program and execution endpoints

### PR DESCRIPTION
**Pull request by facundoolano**
This adds a program execution endpoint, similar to the deployment one. The additional fee calculation (assumed to be the same for both) has been extracted to a helper function. This also fixes a couple of stale comments mentioned in another PR.

This is currently pointing to the #1897 branch, I can send it to the testnet3 branch once that's merged if preferred.

This also incorporates the get program endpoint from #1901

### CODE REVIEW

File: snarkos/ledger/server.rs [hunk 1]
```diff
@@ -196,14 +196,23 @@ impl<N: Network> Server<N> {
             .and(with(ledger_sender.clone()))
             .and_then(Self::transaction_broadcast);
 
-        // POST /testnet3/program/deploy
+        // POST /testnet3/deploy
         let deploy_program = warp::post()
             .and(warp::path!("testnet3" / "deploy"))
```
Comment (raychu86) [inline snarkos/ledger/server.rs]: Let's do "testnet3/program/deploy" and "testnet3/program/execute" when it's convenient to make the switch

File: snarkos/ledger/server.rs [hunk 2]
```diff
@@ -341,17 +351,43 @@ impl<N: Network> Server<N> {
...
+    ) -> Result<impl Reply, Rejection> {
+        let additional_fee = Self::execute_additional_fee(ledger)?;
+        let transaction = Transaction::from_execution(execution.clone(), Some(additional_fee)).or_reject()?;
+        match ledger_sender.send(LedgerRequest::TransactionBroadcast(transaction)).await {
+            Ok(()) => Ok(reply::with_status(reply::json(&json!({ "execution": execution })), StatusCode::OK)),
+            Err(error) => Err(reject::custom(ServerError::Request(format!("{error}")))),
+        }
+    }
+
+    /// Spends the record with the smallest value with at least `1 gate` as the fee
```
Comment (raychu86) [inline snarkos/ledger/server.rs]: "as the fee for the execution of a transaction." can be shortened to "as a transaction fee"

File: snarkos/ledger/server.rs (hunk 1 above)
Comment (facundoolano) [reply snarkos/ledger/server.rs]: sorry do you mean keeping it like this or aligning the endpoint url with the old comment?

Comment (raychu86) [reply snarkos/ledger/server.rs]: I think we should align it to the old comments.

So the deploy endpoint would be "testnet3/program/deploy" and the execute endpoint would be "testnet3/program/execute"

Comment (facundoolano) [reply snarkos/ledger/server.rs]: gotcha, I'll update and talk with the folks to adapt the clients of the endpoint accordingly

File: snarkos/ledger/server.rs [hunk 3]
```diff
@@ -198,12 +198,21 @@ impl<N: Network> Server<N> {
 
         // POST /testnet3/program/deploy
         let deploy_program = warp::post()
-            .and(warp::path!("testnet3" / "deploy"))
+            .and(warp::path!("testnet3" / "program" / "deploy"))
+            .and(warp::body::content_length_limit(10 * 1024 * 1024))
```
Comment (ljedrz) [inline snarkos/ledger/server.rs:210]: do we know the size limit of programs? that would be the best fit here and in `/testnet3/program/execute`

File: snarkos/ledger/server.rs (hunk 1 above)
Comment (facundoolano) [reply snarkos/ledger/server.rs]: updated

File: snarkos/ledger/server.rs (hunk 3 above)
Comment (facundoolano) [reply snarkos/ledger/server.rs:210]: Good question, I actually haven't thought about this, just used the same number as the transaction broadcast and deploy endpoint

Comment (ljedrz) [reply snarkos/ledger/server.rs:210]: the same applies to the other types of data that the protocol constrains, including the transaction :+1:

Comment (facundoolano) [reply snarkos/ledger/server.rs:210]: @raychu86 @howardwu any input on those body lengths?

**Review by raychu86 — APPROVED** [review]

**Pull request merged**